src/plugins/inputs/part.js

```diff
@@ -9,26 +9,36 @@ exports.commands = ["close", "leave", "part"];
 exports.allowDisconnected = true;
 
 exports.input = function(network, chan, cmd, args) {
-	if (chan.type === Chan.Type.LOBBY) {
+	const target = args.length === 0 ? chan : _.find(network.channels, {name: args[0]});
+
+	if (typeof target === "undefined") {
+		chan.pushMessage(this, new Msg({
+			type: Msg.Type.ERROR,
+			text: "Cannot part channel " + target.name + ": Channel not found."
+		}));
+		return;
+	}
+
+	if (target.type === Chan.Type.LOBBY) {
 		chan.pushMessage(this, new Msg({
 			type: Msg.Type.ERROR,
 			text: "You can not part from networks, use /quit instead."
 		}));
 		return;
 	}
 
-	network.channels = _.without(network.channels, chan);
-	chan.destroy();
+	network.channels = _.without(network.channels, target);
+	target.destroy();
 	this.emit("part", {
-		chan: chan.id
+		chan: target.id
 	});
 
-	if (chan.type === Chan.Type.CHANNEL) {
+	if (target.type === Chan.Type.CHANNEL) {
 		this.save();
 
 		if (network.irc) {
 			const partMessage = args[0] ? args.join(" ") : Helper.config.leaveMessage;
```

Comment:
If you provide a target into the command, part message will be wrong here?

Reply:
Oh yeah, good point.

https://en.wikipedia.org/wiki/List_of_Internet_Relay_Chat_commands#PART

Spec says `PART <channels> [<message>]` (where channels is a csv). So I guess we want to check if the chan exists, the nthe part message is the rest of the args. If the chan doesn't exist, then it's a part message. Although that could get messy.

How do other clients deal with this? Do they support `/part` without a chan argument?

Reply:
Mmh does the lobby chan have a name? If not, it will never by `/part`ed from anywhere but itself.

Reply:
It does, the same name you see in the UI.

Reply:
Ok, but what do you expect to happen when somebody does a `/part Freenode` for instance? For know, it responds `You cannot part from networks`, which does not seem unreasonable to me. Though, I'm ready to change this behavior if needed.

Reply:
Oh well I missed the point about the part message. I just tested in weechat: it checks that the first argument starts with a `#`. If so, it parts this chan. Otherwise, it considers the argument to be the part message. The drawback is that it prevents one from parting a private chan (`/part SomeNickname` does not work and just parts the current window).

Reply:
That sounds like a sensible way to do things, I think.

Reply:
You could just check if first argument is CHANNEL or QUERY and it'll work just fine.

Reply:
There's a problem with that, @xPaw 

Say I do `/part thelounge is annoying me` or something similar, then we check if `thelounge` is a channel, and it is, how do we know if the user wants to part that channel, or whether it's part of the part message.

Reply:
There's not much we can do about that. But channels do have to be a full name (so `#thelounge` in this case).

Reply:
If they have to include the `#`, then I think that's fine. I wouldn't want it to work without the `#`, otherwise we could get into some problems.

Reply:
The problem mainly concerns query windows. If I `/part YaManicKill is annoying me` while I'm looking at the `#thelounge` window, but have a query window openned in background with YaManicKill, which one should I quit?

Reply:
Well, a part message makes no sense for a query window.

So I'd say that if there is more than 1 arg and we are trying to part a query, don't let the user do it.

Reply:
I propose the following:

 * `/part #foo bar` quits `#foo` with message "bar".
 * `/part foo bar` quits the **current window** with message "foo bar" even if "foo" is a valid query window..
 * `/part foo` with "foo" a valid query window quits the query window `foo` with no message.
 * `/part foo` with "foo" **not** being a valid query window quits the current window with message "foo".

As a by-product, parting a query window with a leave message is still possible, but only by switching to this window and `/part some message`-ing.

Reply:
Sounds fine to me.

> parting a query window with a leave message is still possible, 

Since part is only sent for channels, it's effectively discarded.

Reply:
Yup, sounds fine to me also.

Reply:
also, `/part #foo bar` displays an error message and do nothing if `#foo` does not exist as a window. It does not part the current window as it would do for `/part foo bar` (with no `#`)